File: motor/motor.py
# motor parameters
K = 1
TS = 0.01
TC = 1.2
A1 = K * TS / (2 * TC + TS)
A2 = (TS - 2 * TC) / (2 * TC + TS)


def motor(input, prev_input, prev_output):
    return ((input + prev_input) * A1 - prev_output * A2)

File: motor/motor.c
#include "motor.h"

float motor(float input, float prev_input, float prev_output)
{
    float res = 0;

    res = ((input + prev_input) * K * TS - prev_output * (TS - 2 * TC)) / (2 * TC + TS);

    return res;
}